src/gondi.ts

```diff
@@ -1067,6 +1073,121 @@ export class Gondi {
   } & Parameters<Contracts['UserVaultV5']['burnAndWithdraw']>[0]) {
     return this.contracts.UserVault(userVaultAddress).burnAndWithdraw(data);
   }
+  async wrapOldERC721({
+    collection,
+    tokenId,
+  }: {
+    collection: {
+      id: string;
+      wrapperCollections?: { contractData: { contractAddress: Address } }[];
+    };
+    tokenId: bigint;
+  }) {
+    if (collection?.id === undefined) throw Error('Invalid collection');
+
+    const wrapperAddress = await this._getWrapperAddress(collection);
+    if (wrapperAddress === undefined) throw Error('Collection has no associated wrappers');
+
+    const wrapper = this.contracts.OldERC721Wrapper(wrapperAddress);
+    const wrappedAddress = await wrapper.contract.read.wrapped();
+    const wrapped = this.contracts.OldERC721(wrappedAddress);
+    const stashAddress = await wrapper.contract.read.stashAddress([this.wallet.account.address]);
+    const txTransfer = await wrapped.safeContractWrite.transfer([stashAddress, tokenId]);
+
+    const waitTransferMined = async () => {
+      const receipt = await this.bcClient.waitForTransactionReceipt({
+        hash: txTransfer,
+      });
+      return receipt;
+    };
+
+    const waitMintMined = async () => {
+      const receipt = await this.bcClient.waitForTransactionReceipt({
+        hash: await txMint,
+      });
+      return receipt;
+    };
+
+    const waitMined = async () => {
+      return [await waitTransferMined(), await waitMintMined()];
+    };
+
+    const txMint: Promise<Address> = new Promise(async (resolve, _reject) => {
+      await waitTransferMined();
+      resolve(await wrapper.safeContractWrite.wrap([tokenId]));
+    });
+    return {
+      waitTransferMined,
+      waitMintMined,
+      waitMined,
+    };
+  }
```

Comment:
Moving all these to a file in contracts makes more sense to me, so that we keep separating blockchain interaction logic in different files.

Reply:
I'd return one function which does everything and that's it